gunicorn/config.py

```diff
@@ -344,6 +344,20 @@ def validate_pos_int(val):
     return val
 
 
+def validate_ssl_version(val):
+    ssl_versions = {}
+    for protocol in [p for p in dir(ssl) if p.startswith("PROTOCOL_")]:
+        ssl_versions[protocol[9:]] = getattr(ssl, protocol)
+    if val in ssl_versions:
+        return ssl_versions[val]
+    else:
+        intval = validate_pos_int(val)
+        if intval in ssl_versions.values():
+            return intval
+
+    raise ValueError("Invalid ssl_version: %s" % val)
```

Comment:
What do you think about making this exception message more user friendly by printing available choices?

For example:

```py
'Invalid value for ssl_version: %s. Available choices: %s'
```

I'm not a native speaker, so feel free to suggest better wording :)

Also, it would be nice if we could warn users if they passed a `val` that starts with `PROTOCOL`.

Reply:
Good idea, done.  Since we now supply the valid options in the error, I suppose a separate warning is not needed for values starting with `PROTOCOL_`.

```
Invalid ssl_version: 99991234. Valid options: TLSv1,TLSv1_2,SSLv3,SSLv23,TLSv1_1
```